react-effects/my-app/src/List.js

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable no-unused-vars -- REMOVE ME */
+import { useEffect, useState } from 'react';
+import readItems from './read';
+
+export default function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState([]);
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const response = await readItems();
+        setItems(response);
+        setIsLoading(false);
+      } catch (err) {
+        setError(err);
```

Comment:
If an error happens, isLoading will still be true so you'll display the loading message instead of the error message. You need to set isLoading to false after an error, too.

The easiest way to do this is to just call `setIsLoading(false);` inside the `finally` clause of the `try/catch` block.